bench/workgen/workgen.cxx

```diff
@@ -415,31 +415,16 @@ Monitor::~Monitor() {}
 int Monitor::run() {
     struct timespec t;
     struct tm *tm, _tm;
-    char time_buf[64], version[100];
+    char version[100];
     Stats prev_totals;
     WorkloadOptions *options = &_wrunner._workload->options;
     uint64_t latency_max = (uint64_t)options->max_latency * THOUSAND;
-    size_t buf_size;
-    bool first;
+    bool first_iteration;
 
-    (*_out) << "#time,"
-            << "totalsec,"
-            << "read ops per second,"
-            << "insert ops per second,"
-            << "update ops per second,"
-            << "checkpoints,"
-            << "read average latency(uS),"
-            << "read minimum latency(uS),"
-            << "read maximum latency(uS),"
-            << "insert average latency(uS),"
-            << "insert min latency(uS),"
-            << "insert maximum latency(uS),"
-            << "update average latency(uS),"
-            << "update min latency(uS),"
-            << "update maximum latency(uS)"
-            << std::endl;
+    // Format header of the table in _out stream.
+    _format_out_header();
```

Comment:
Should this class take a ostream as an arg and become static ?

Reply:
If we decide to go that path, we could do the same with _format_json_prefix and the other methods where applicable.

Reply:
Good point. Ideally, all the formatting functionality should be removed from Monitor. There should be output formatter interface and two classes that implement it: JsonOutputFormatter and TextOutputFormatter. Monitor class would instantiate and initialise required formatters for that particular run and call interface methods before run, after each iteration and after the run is finished. With that implementation it would be super easy to add another formatter if we need to. Not sure if that's an overkill. What do you think?

Reply:
That would be better indeed, an interface OutputFormatter and then derived classes depending on the output. It is beyond this scope though, a new ticket should be created to suggest this improvement.